app/components/home/Challenge.tsx

```diff
@@ -108,6 +111,9 @@ export function Challenge({
   };
 
   const handleDoubleTap = () => {
+    if (currentUser.name === "Guest") {
```

Comment:
I understand this is made for guest users to not be able to like a post. Wouldn't it be better to centralize it in handleLikePress which is called by the other one ?

Reply:
It's not the same thing ! One is for pressing the like button, the other one for when you double tap on the post and it likes.

Reply:
I understand but I'm talking about restraining the guest user. As the DoubleTap calls the handleLike, maybe you could retrain the user in handleLikePress to avoid duplicate code.